.github/workflows/codeql-analysis.yml

```diff
@@ -22,8 +22,28 @@ on:
     - cron: '39 8 * * 4'
 
 jobs:
+  # Check that the patches apply cleanly as quickly as possible. This means we don't waste machine
+  # time running CodeQL setup and analysis when the build can't possibly succeed.
+  #
+  # This also happens to be a good signal for devs. GitHub Actions is quick to get an agent, so when
+  # this job fails, it's easy to see, a nd clear that the rest of the PR's jobs aren't going to
+  # succeed. (The actual tests run in AzDO, so we can't cancel them from here, but at least the dev
+  # will know the cause of the failures without navigating through the links.)
+  check_patches:
+    name: Patches Apply Cleanly
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - run: pwsh eng/run.ps1 submodule-refresh -shallow
+
   analyze:
     name: Analyze
+    # The instrumented build will fail if the patches don't apply cleanly, so don't. This saves
```

Comment:
`so don't` is there a verb missing?

Reply:
I meant for it to refer to "build" in some way, but yeah, that doesn't work. 😄